src/pseudopeople/entity_types.py

```diff
@@ -67,10 +67,15 @@ def __call__(
         randomness_stream: RandomnessStream,
         additional_key: Any,
     ) -> pd.Series:
+        if column.empty:
+            return column
         column = column.copy()
         noise_level = configuration[
             Keys.CELL_PROBABILITY
-        ] * self.noise_level_scaling_function(column.name)
+        ] * self.noise_level_scaling_function(column)
+        # Certain columns have a their noise level scaled so we must check to make sure the noise level is within the
+        # allowed range between 0 and 1 for probabilities
+        noise_level = min(noise_level, 1.0)
         to_noise_idx = get_index_to_noise(
```

Comment:
So noise_level is not a series, correct? And `get_index_to_noise` just worked fine? At the very least, change the typing (still says float)

Reply:
Yes it is a float.  The type hin in `get_index_to_noise `says it should be a float.  Which one are you referring to?